gogs_client/interface.py

```diff
@@ -42,6 +42,68 @@ def authenticated_user(self, auth):
         response = self._get("/user", auth=auth)
         return GogsUser.from_json(self._check_ok(response).json())
 
+    def get_tokens(self, auth, username=None):
+        """
+        Returns tokens defined for specified user.
+        If no user specified uses user authenticated by the given authentication.
+        Right now, authentication must be UsernamePassword (not Token).
+
+        :param auth.Authentication auth: authentication for user to retrieve
+        :param str username: username of owner of tokens
+
+        :return: list of token info representation
+        :rtype: List[TokenInfo]
+        :raises NetworkFailure: if there is an error communicating with the server
+        :raises ApiFailure: if the request cannot be serviced
+        """
+        if username is None:
+            username = self.authenticated_user(auth).username
+        response = self._get("/users/{u}/tokens".format(u=username), auth=auth)
+        return [TokenInfo.from_json(o) for o in self._check_ok(response).json()]
+
+    def create_token(self, auth, name, username=None):
+        """
+        Creates new token with specified name for specified user.
+        If no user specified uses user authenticated by the given authentication.
+        Right now, authentication must be UsernamePassword (not Token).
+
+        :param auth.Authentication auth: authentication for user to retrieve
+        :param str name: name of new token
+        :param str username: username of owner of new token
+
+        :return: representation of token info
+        :rtype: TokenInfo
+        :raises NetworkFailure: if there is an error communicating with the server
+        :raises ApiFailure: if the request cannot be serviced
+        """
+        if username is None:
+            username = self.authenticated_user(auth).username
+        data = {"name": name}
+        response = self._post("/users/{u}/tokens".format(u=username), auth=auth, data=data)
+        return TokenInfo.from_json(self._check_ok(response).json())
+
+    def ensure_token(self, auth, name, username=None):
+        """
+        Creates new token if token with specified name for specified user does not exists.
+        If no user specified uses user authenticated by the given authentication.
+        Right now, authentication must be UsernamePassword (not Token).
+
+        :param auth.Authentication auth: authentication for user to retrieve
+        :param str name: name of new token
+        :param str username: username of owner of new token
+
+        :return: token authenticator
+        :rtype: Token
```

Comment:
As things currently stand, shouldn't this be `TokenInfo`? For instance, this method could return `self.create_token(..)`, which is of type `TokenInfo`.

I still would suggest scraping the `TokenInfo` class altogether (see other comment), but want to make sure I'm not missing something.